src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -267,6 +267,7 @@ export class HttpConnection implements IConnection {
                         // the returned access token
                         const accessToken = negotiateResponse.accessToken;
                         this._accessTokenFactory = () => accessToken;
+                        this._httpClient.replaceAccessTokenFactory(this._accessTokenFactory);
```

Comment:
Why not just set `this._httpClient._accessToken = accessToken`. That way the `AccessTokenHttpClient` doesn't make a second request with the same access token for no reason.